### PR TITLE
Add support to importing a project by projectnumber

### PR DESCRIPTION
**Pull request by spew**
Some google services, such as Cloud Asset Inventory, use the ProjectNumber as the ID of the project instead of the project id. This change adds support to importing by Number to the Project resource.

### CODE REVIEW

**Comment by emilymye** [comment]
Do you mind adding a test to resource_google_project_test.go?  You can add a test step (or 2) to  TestAccProject_create that uses the (Import step format)[https://github.com/GoogleCloudPlatform/magic-modules/blob/4540f9c4260c86d6d7361641097a18255da00228/third_party/terraform/tests/resource_google_project_test.go#L142-L147] - ideally we'd want one with `ImportStateId : "projects/" + projectNumber` and one with `ImportStateId : "projects/" + pid` to be explicit

File: google/resource_google_project.go [hunk 1]
```diff
@@ -396,18 +399,19 @@ func resourceGoogleProjectDelete(d *schema.ResourceData, meta interface{}) error
...
 	if matched {
-		return nil, fmt.Errorf("Error importing project %q, please use project_id", pid)
+		// this is actually a project number not a project id
+		d.Set("number", pid)
+	} else {
+		d.Set("project_id", pid)
 	}
-
 	// Ensure the id format includes projects/
-	d.SetId(fmt.Sprintf("projects/%s", pid))
```
Comment (emilymye) [inline google/resource_google_project.go:410]: Can you add a comment here and at the top of Read() saying that project ID might be either project ID or number?

**Comment by spew** [comment]
Yes -- I will do so

**Comment by c2thorn** [comment]
Hey @spew , we are cleaning up some lingering PR's. Is this something you are still intending to work on? Otherwise we can probably close after a few days.

**Comment by spew** [comment]
Hi, yes, I would like to get this in. I am curious if unit tests are a standard requirement for all importers?

**Comment by rileykarson** [comment]
Closing PR as stale

**Pull request closed without merging**